### ISSUE TITLE
Tabs being closed automatically

### ISSUE THREAD

**Issue by juvogel** [uBlock-LLC/uBlock#1051]
Running version 0.9.1.1

When I click on some links it automatically closes the tab. When I disable ublock thats when this behavior stops.

Example link:
http://ib.adnxs.com/clktrb?id=386368
This link does not close if pasted directly into the browser but when clicked from a link it closes. (Link originally on the Microsoft website, sliding gallery).

**Comment by gorhill** [comment]
Browser/version? Etc. ("Please review the [guidelines for contributing](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md) to this repository")

**Comment by mlindner** [comment]
I'm also experiencing this.

URL https://www.reddit.com/r/WTF/comments/2zlf1h/the_strangest_constipation_ad_youre_going_to_see/
If I visit this URL by clicking on it from reddit's front page then it closes the tab. Visiting this URL normally does not close the tab.

Google Chrome Version 41.0.2272.89 (64-bit) on Mac OS X 10.8.5

**Comment by alphabt** [comment]
Same here with above two links.

Chrome 41.0.2272.101 (64-bit) on OS X 10.10.2

**Comment by gorhill** [comment]
"Please review the [guidelines for contributing](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md) to this repository"

**Comment by mlindner** [comment]
Your requested information has been given. What is missing? I only started using ublock yesterday, if the plugin is this buggy and support this bad I'm going to be switching back to Adblock for a while. I just had another unrelated page close on me.

**Comment by offhub** [comment]
What is missing?
- Preferences Different From Defaults
  We recommend a screenshot — include any filter lists you enabled/disabled, whitelisted sites, etc.
- Other Extensions

**Comment by kurtextrem** [comment]
I can confirm this. Disable uBlock on google search results, google after "superdry" and click the text ad. It redirects to the url and closes the tab immediately.

**Comment by gorhill** [comment]
@kurtextrem I can repro. Investigating.

**Comment by gorhill** [comment]
Fixed #510 with 12d2ab526fad63f38004fa906a5a8d0b61cc1eba. As far as I can tell this fixes https://github.com/gorhill/uBlock/issues/1051#issuecomment-84053855 ("superdry" case).

I wasn't able to reproduce the other cases so I can't tell if 12d2ab526fad63f38004fa906a5a8d0b61cc1eba fixes them.

**Comment by ghost** [comment]
I have the same problem with the latest dev build. When I try to open a link in a new tab (in background), the new tab close automatically. It happen, when I try to open a link from reddit to imgur (not all of them), for example: http://imgur.com/a/0jg9d

**Version:** 0.9.2.0-rc.0
**Browser:** Opera 29.0.1795.14 (dev)
**OS:** Windows 7 64-bit
**Other extensions:** BetterTTV, Download Chrome Extension, F.B. Purity for Facebook, Google Dictionary (by Google), Imagus, LastPass, Reddit Enhancement Suite

I'm using the default filters list + EasyList Liste FR‎ and the dynamic filtering (Default-deny).

**Comment by my-password-is-password** [comment]
I can reproduce.  Using @mlindner reddit link posted above.  Middle click it.

It automatically closes the new tab.  I'm using uBlock `0.9.2.0` on chromium dev build.

Selected lists:

```
 Anti-Adblock Killer | Reek
 Fanboy+Easylist-Merged Ultimate List‎
```

**Comment by gorhill** [comment]
@chway I can reproduce. According to the request log, there are many redirections, and one of them is `http://imgur.com/ads?page=inside&main=1&hot=1&time=1426876583`, which matches the filter `.com/ads?$popup`.

@my-password-is-password I can reproduce too. Just to be clear I could not reproduce before 0.9.2.0.

The new code also listen to embedded frames, in order to pass all the tests in #510, I suspect it's what is happening.

**Comment by gorhill** [comment]
Fixed with 04a5786c41d830bf787369a16efb6efaea6685c8.
